code/modules/chemistry/Reagents-Misc.dm

```diff
@@ -221,40 +221,53 @@ datum
 			fluid_r = 120
 			fluid_g = 255
 			fluid_b = 240
 			transparency = 215
+			var/theraputicamount = 5
 			value = 41 // 17 18 6
 			viscosity = 0.4
 
-			on_add()
-				if (istype(holder) && istype(holder.my_atom) && hascall(holder.my_atom,"add_stam_mod_regen"))
-					holder.my_atom:add_stam_mod_regen("aranesp", 15)
-				if (istype(holder) && istype(holder.my_atom) && hascall(holder.my_atom,"add_stam_mod_max"))
-					holder.my_atom:add_stam_mod_max("aranesp", 25)
+			on_add(var/mob/M)
+				M = holder?.my_atom
+				if(istype(M) && holder.get_reagent_amount(src.id) > theraputicamount)
+					M.add_stam_mod_regen("aranesp", 15)
+					M.add_stam_mod_max("aranesp", 25)
 				return
 
-			on_remove()
-				if (istype(holder) && istype(holder.my_atom) && hascall(holder.my_atom,"remove_stam_mod_regen"))
-					holder.my_atom:remove_stam_mod_regen("aranesp")
-				if (istype(holder) && istype(holder.my_atom) && hascall(holder.my_atom,"remove_stam_mod_max"))
-					holder.my_atom:remove_stam_mod_max("aranesp")
+			on_remove(var/mob/M)
+				M = holder?.my_atom
+				if(istype(M))
+					M.remove_stam_mod_regen("aranesp")
+					M.remove_stam_mod_max("aranesp")
 				return
 
 			on_mob_life(var/mob/M, var/mult = 1)
-				if (!M) M = holder.my_atom
-				if (prob(90))
-					M.take_toxin_damage(1 * mult)
-				if (prob(5)) M.emote(pick("twitch", "shake", "tremble","quiver", "twitch_v"))
-				if (prob(8)) boutput(M, "<span style=\"color:blue\">You feel [pick("really buff", "on top of the world","like you're made of steel", "food_energized", "invigorated", "full of energy")]!</span>")
-				if (prob(5))
-					boutput(M, "<span style=\"color:red\">You cannot breathe!</span>")
-					M.setStatus("stunned", max(M.getStatusDuration("stunned"), 20 * mult))
-					M.take_oxygen_deprivation(15 * mult)
-					M.losebreath += (1 * mult)
-				..()
+				M = holder?.my_atom
+				if(istype(M))
+					if(holder.get_reagent_amount(src.id) > theraputicamount)
+						if (prob(90))
+							M.take_toxin_damage(1 * mult)
+						if (prob(5))
+							M.emote(pick("twitch", "shake", "tremble","quiver", "twitch_v"))
+						if (prob(8))
+							boutput(M, "<span style=\"color:blue\">You feel [pick("really buff", "on top of the world","like you're made of steel", "food_energized", "invigorated", "full of energy")]!</span>")
+						if (prob(5))
+							boutput(M, "<span style=\"color:red\">You cannot breathe!</span>")
+							M.setStatus("stunned", max(M.getStatusDuration("stunned"), 20 * mult))
+							M.take_oxygen_deprivation(15 * mult)
+							M.losebreath += (1 * mult)
+					else
+						M.remove_stam_mod_regen("aranesp")
+						M.remove_stam_mod_max("aranesp")
+						if(ishuman(M))
+							var/mob/living/carbon/human/H = M
+							if(blood_system)
+								H.blood_volume += 1  * mult
+							H.nutrition -= 1  * mult
```

Comment:
nutrition is a var on mobs and not humans, bring out of scope *of the human check*
Then, bring the human check back before the blood system check in the same if and cast below, like you had previously.